react-accordion-component/my-app/src/Accordion.jsx

```diff
@@ -0,0 +1,43 @@
+import './Accordion.css';
+import { useState } from 'react';
+
+export default function Accordion({ listOfTopics }) {
+  const [current, setCurrent] = useState('');
+
+  function handleCustomClick(clickedButton) {
+    if (clickedButton === current) {
+      setCurrent('');
+    } else {
+      setCurrent(clickedButton);
+    }
+  }
+
+  const topicsArray = [];
+  for (let i = 0; i < listOfTopics.length; i++) {
+    topicsArray.push(
+      <Topic
+        key={listOfTopics[i].id}
+        currentClicked={current}
+        onCustomClick={handleCustomClick}
+        item={listOfTopics[i]}
+      />
+    );
+  }
+
+  return <div>{topicsArray}</div>;
+}
+
+function Topic({ item, onCustomClick, currentClicked }) {
+  return (
+    <div>
+      <button
+        onClick={(e) => onCustomClick(e.target.textContent)}
```

Comment:
This doesn't really fix the issue Tim was worried about since it still passes `e.target.textContent`. The best way to handle cases like this is to just pass `item` to `onCustomClick`.